Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy.swift

```diff
@@ -105,10 +105,32 @@ struct PathHierarchy {
             if let language {
                 moduleNode.languages.insert(language)
             }
-            
+
+            var symbols = graph.symbols
+            var relationships = graph.relationships
+
+            // Overload group symbols are added directly to the unified graph, not any individual
+            // graphs, so load up any relevant symbols and relationships before continuing
+            if let unifiedSelector = UnifiedSymbolGraph.Selector(forSymbolGraph: graph),
+               let unifiedGraph = loader.unifiedGraphs[moduleNode.name],
+               !unifiedGraph.overloadGroupSymbols.isEmpty
+            {
+                for overloadGroupIdentifier in unifiedGraph.overloadGroupSymbols {
+                    if let unifiedSymbol = unifiedGraph.symbols[overloadGroupIdentifier],
+                       let overloadSymbol = SymbolGraph.Symbol(fromUnifiedSymbol: unifiedSymbol, selector: unifiedSelector) {
+                        symbols[overloadGroupIdentifier] = overloadSymbol
+                    }
+                }
+
+                let overloadRelationships = unifiedGraph.relationshipsByLanguage[unifiedSelector]?.filter({ relationship in
+                    unifiedGraph.overloadGroupSymbols.contains(relationship.source) || unifiedGraph.overloadGroupSymbols.contains(relationship.target)
+                }) ?? []
+                relationships.append(contentsOf: overloadRelationships)
+            }
```

Comment:
Rather than mixing symbol graph data and unified symbol graph data. Can we add the overload groups after all the symbol graphs (outside this loop)? That would allow for all the overload specific logic to move into one place.

I can't checkout that branch of SymbolKit, so I couldn't test this but I think the only thing that the path hierarchy needs to do with overload groups are:
- Disfavor the overloaded symbols in the overload group
- Add a node for the overload group in the same location as the overloaded symbols

If so, we should be able to do that using something like this:

```swift
// Overload group don't exist in the individual symbol graphs.
// Since overload groups don't change the _structure_ of the path hierarchy, we can add them after after all symbols for all platforms have already been added.
for unifiedGraph in loader.unifiedGraphs.values {
    // Create nodes for all the overload groups
    let overloadGroupNodes: [String: Node] = unifiedGraph.overloadGroupSymbols.reduce(into: [:]) { acc, uniqueID in
        assert(allNodes[uniqueID] == nil,
               "Overload group ID \(uniqueID) already has a symbol node in the hierarchy: \(allNodes[uniqueID]!.map(\.name).sorted().joined(separator: ","))")
        guard let unifiedSymbol = unifiedGraph.symbols[uniqueID] else { return }
        guard let symbol = unifiedSymbol.defaultSymbol else {
            fatalError("Overload group \(uniqueID) doesn't have a default symbol.")
        }
        acc[uniqueID] = Node(symbol: symbol, name: symbol.pathComponents.last!)
    }
    
    for relationship in unifiedGraph.relationships where relationship.kind == .overloadOf {
        guard let groupNode = overloadGroupNodes[relationship.source], let overloadedSymbolNodes = allNodes[relationship.target] else {
            continue
        }
        
        for overloadedSymbolNode in overloadedSymbolNodes {
            // We want to disfavor the individual overload symbols in favor of resolving links to their overload group symbol.
            overloadedSymbolNode.specialBehaviors.formUnion([.disfavorInLinkCollision, .excludeFromAutomaticCuration])
            
            guard let parent = overloadedSymbolNode.parent else { continue }
            
            assert(groupNode.parent == nil || groupNode.parent === parent, """
            Unexpectedly grouped symbols with different locations in the symbol hierarchy:
            Group ID: \(groupNode.symbol!.identifier.precise)
            Locations: \(Set(overloadedSymbolNodes.map { $0.symbol!.pathComponents.joined(separator: "/") }.sorted()))
            """)
            parent.add(symbolChild: groupNode)
        }
        assert(groupNode.parent != nil, "Unexpectedly found no location in the hierarchy for overload group \(relationship.source)")
    }
}
```

Reply:
I started trying to do this, but i wound up trying to reimplement the single-symbol logic instead. I'll try your version and see if that works.

Reply:
> I can't checkout that branch of SymbolKit, so I couldn't test this

Whoops, i forgot to update DocC's Package.resolved after i amended my commit. 😅 You should be able to run `swift package update swift-docc-symbolkit` to get a valid commit. I'll push that change here once i get your code worked in.